products/statement-generator/src/tests/textareas.test.tsx

```diff
@@ -0,0 +1,115 @@
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+
+import { ThemeProvider } from '@material-ui/core/styles';
+import customMuiTheme from 'styles/customMuiTheme';
+
+import Textarea from '../components/Textarea';
+
+describe('Textarea component', () => {
+  test('Textarea renders and displays correct default value', () => {
+    const { getByText } = render(
+      <ThemeProvider theme={customMuiTheme}>
+        <Textarea id="test" handleChange={() => {}} defaultValue="Type here" />
+      </ThemeProvider>
+    );
+
+    const textarea = getByText(/Type here/i);
+
+    expect(textarea).toBeInTheDocument();
+  });
+
+  test('Textarea props are passed correctly', () => {
+    interface ExpectedProps {
+      placeholder: string;
+      id: string;
+      rows: string;
+    }
+
+    const checkTextareaProps = (
+      input: HTMLElement,
+      expectedProps: ExpectedProps
+    ) => {
+      const attributes: (keyof typeof expectedProps)[] = [
+        'placeholder',
+        'rows',
+        'id',
+      ];
+      attributes.forEach((attr) => {
+        expect(input).toHaveAttribute(attr, expectedProps[attr]);
+      });
+    };
+
+    const { getByText, rerender } = render(
+      <ThemeProvider theme={customMuiTheme}>
+        <Textarea
+          id="test"
+          label="label"
+          placeholder="Type here"
+          handleChange={() => {}}
+          defaultValue="Type here"
+          rows={3}
+        />
+      </ThemeProvider>
+    );
+
+    const textarea = getByText(/Type here/i);
+
+    // passed correct placeholder, # of rows, and id
+    checkTextareaProps(textarea, {
+      placeholder: 'Type here',
+      rows: '3',
+      id: 'test',
+    });
+
+    // passed correct label
+    const label = getByText('label');
+    expect(label).toBeInTheDocument();
+    expect(label).toHaveAttribute('for', 'test');
+
+    // passed disabled state correctly
+    rerender(
+      <ThemeProvider theme={customMuiTheme}>
+        <Textarea
+          id="test"
+          label="label"
+          handleChange={() => {}}
+          defaultValue="Type here"
+          rows={3}
+          disabled
+        />
+      </ThemeProvider>
+    );
+
+    expect(textarea).toBeDisabled();
+  });
+
+  test('Textarea is accessible and focusable', () => {
+    const { getByText, getByPlaceholderText } = render(
+      <ThemeProvider theme={customMuiTheme}>
+        <Textarea
+          id="accessible-textarea"
+          label="accessibleTextarea"
+          placeholder="Type here"
+          handleChange={() => {}}
+        />
+      </ThemeProvider>
+    );
+
+    // label is associated with the textarea
+    const label = getByText('accessibleTextarea');
+    expect(label).toBeInTheDocument();
+    expect(label).toHaveAttribute('for', 'accessible-textarea');
+
+    // textarea is focusable
+    const input = getByPlaceholderText('Type here');
+    userEvent.click(input);
+    expect(input).toHaveFocus();
```

Comment:
There are other ways to focus on a text area, like tabbing from the a different element. Our tests should cover that as well.